packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -174,18 +181,20 @@ function getEndpoint(tableId: string | undefined, operation: string) {
   }
 }
 
-function basicProcessing(row: Row, table: Table): Row {
+function basicProcessing(row: Row, table: Table, isLinked: boolean): Row {
   const thisRow: Row = {}
   // filter the row down to what is actually the row (not joined)
-  for (let fieldName of Object.keys(table.schema)) {
+  for (let field of Object.values(table.schema)) {
+    const fieldName = field.name
+
     const pathValue = row[`${table.name}.${fieldName}`]
-    const value = pathValue != null ? pathValue : row[fieldName]
+    const value = pathValue != null || isLinked ? pathValue : row[fieldName]
```

Comment:
I find the ternary in combination with the `||` quite difficult to read. Could we extract this logic to `if/else` instead?

Question: Is this the same as the logic from `generateRowId`? If so, we could extract out to a shared function?

Reply:
Changed! Good spot